cli/azd/cmd/container.go

```diff
@@ -545,8 +545,8 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 			return nil, fmt.Errorf("getting platform config: %w", err)
 		}
 
-		if !ok {
-			return nil, errors.New("platform config not found")
+		if !ok || platformConfig.Type == "" {
```

Comment:
Given that:

1. We're in the logic that provides the `platform.Config`
2. `platform`, `platform.type` are optional values.

I wonder if it's preferable for a function that returns `platform.Config` to return suitable values (`nil`/empty struct, empty type) that reflects this. We should still honor contract invariants as we do to validate for correctness, but these things are truly optional.

Don't mind the current approach as well, but I do wonder if we end up creating layers here for internal abstraction that creates slight complexity.

Reply:
The intent here is to either return a valid `platform.Config` instance or return an error.  A `platform.Config` instance is not valid unless `Type` is set to a valid value of `devcenter` or `default`